torch/optim/lr_scheduler.py

```diff
@@ -219,21 +240,24 @@ class MultiStepLR(_LRScheduler):
     """
 
     def __init__(self, optimizer, milestones, gamma=0.1, last_epoch=-1):
-        if not list(milestones) == sorted(milestones):
-            raise ValueError('Milestones should be a list of'
-                             ' increasing integers. Got {}', milestones)
-        self.milestones = milestones
+        self.milestones = Counter(milestones)
         self.gamma = gamma
         super(MultiStepLR, self).__init__(optimizer, last_epoch)
 
     def get_lr(self):
+        if self.last_epoch not in self.milestones:
+            return [group['lr'] for group in self.optimizer.param_groups]
+        return [group['lr'] * self.gamma ** self.milestones[self.last_epoch]
+                for group in self.optimizer.param_groups]
+
+    def get_lr_closed_form(self):
         return [base_lr * self.gamma ** bisect_right(self.milestones, self.last_epoch)
```

Comment:
This still needs to get fixed right?

Reply:
Actually, now I'm surprised the tests are passing. What happened here?

Reply:
The issue was that the learning rate around line 84 was computed before the epoch was incremented. That has been fixed above.